code/obj/item/kitchen.dm

```diff
@@ -766,10 +766,29 @@ TRAYS
 		if (isitem(a) && can_reach(user, src) && can_reach(user, a))
 			src.add_contents(a, user, params2list(params))
 
-	attack_self(mob/user) // in case you only have one arm or you stacked too many MONSTERs or something just dump a random piece of food
+	// in case you only have one arm or you stacked too many MONSTERs or something just dump a random piece of food
+	// chefs are too fancy for that and will instead get to name the dish
+	attack_self(mob/user)
 		. = ..()
-		if (length(src.contents))
-			src.remove_contents(pick(src.contents))
+		if(user.traitHolder?.hasTrait("training_chef"))
+			tooltip_rebuild = 1
+			var/holder = src.loc
+			var/str = copytext(html_encode(tgui_input_text(user, "Dish name?", "Set name")), 1, 32)
```

Comment:
`tgui_input_text` takes a `max_length` argument for preventing the user from entering text past a certain length (you should still check again on the server though)